packages/graph/hash_graph/lib/graph/src/api/rest/mod.rs

```diff
@@ -109,3 +141,43 @@ impl Modify for MergeAddon {
         );
     }
 }
+
+/// Addon to allow external references in schemas.
+///
+/// Any component that starts with `EXTERNAL_` will transform into a relative URL in the schema and
+/// receive a `.json` ending.
+///
+/// For example the `EXTERNAL_Entity` component will be transformed into `./models/Entity.json`
+struct ExternalRefAddon;
+
+impl Modify for ExternalRefAddon {
+    fn modify(&self, openapi: &mut openapi::OpenApi) {
```

Comment:
Self requesting changes here: But we want to traverse components as well and do this substitution

Reply:
Addressed in 1c70848b5493fd4705518f85e4ca40ac420ccdb6